src/server/request.rs

```diff
@@ -66,17 +68,31 @@ impl<'a, T> Request<'a, T> {
     #[inline]
     pub fn version(&self) -> &HttpVersion { &self.version }
 
-    /*
     /// The target path of this Request.
     #[inline]
-    pub fn path(&self) -> Option<&str> {
-        match *self.uri {
-            RequestUri::AbsolutePath(ref s) => Some(s),
-            RequestUri::AbsoluteUri(ref url) => Some(&url[::url::Position::BeforePath..]),
-            _ => None
+    pub fn path(&self) -> Option<Cow<str>> {
+        match self.uri {
+            RequestUri::AbsolutePath(ref s) => {
+                let url = Url::parse("http://example.com").ok().and_then(|u| u.join(s).ok());
```

Comment:
I was going for consistency and doing it in a way that I don't have to read the RFC and implement correct parsing and interpretation of urls. Url::parse normalizes the urls, and therefore, it makes a copy. It would be weird if normalized paths and queries were returned for AbsoluteUri but not for AbsolutePath.